pkg/ossm/smcp_annotation.go

```diff
@@ -89,18 +89,31 @@ func TestSMCPAnnotations(t *testing.T) {
 			util.KubeApplyContents("bookinfo", testAnnotationProxyEnv)
 		}
 		util.CheckPodRunning("bookinfo", "app=env")
-		time.Sleep(time.Duration(10) * time.Second)
-
-		msg, _ := util.Shell(`kubectl get po -n bookinfo -l app=env -o yaml | grep "annotation-from-smcp"`)
-
-		if !strings.Contains(msg, `test1.annotation-from-smcp: test1`) {
-			t.Errorf("Failed to get annotations: %v", `test1.annotation-from-smcp: test1`)
+		podName, err := util.GetPodName("bookinfo", "app=env")
+		if err != nil {
+			t.Fatalf("Failed to get pod name: %v", err)
+		}
+		annotations, err := util.GetPodAnnotations("bookinfo", podName, 30)
+		if err != nil {
+			t.Fatalf("Failed to get annotations: %v", err)
+		}
+		util.Log.Infof("Checking annotation: %", annotations["test1.annotation-from-smcp"])
+		if _, ok := annotations["test1.annotation-from-smcp"]; !ok {
+			t.Errorf("Failed to get annotations: test1.annotation-from-smcp: test1")
+		} else if annotations["test1.annotation-from-smcp"] != "test1" {
+			t.Errorf("Failed to get annotations: test1.annotation-from-smcp: test1")
```

Comment:
This is totally up to you, but I think that it would be easier to understand why the test failed if you changed it as follows:
```suggestion
			t.Errorf("Failed to get annotations: test1.annotation-from-smcp: expected 'test1', got '%s'", )
```